db/db_impl_write.cc

```diff
@@ -62,6 +62,11 @@ Status DBImpl::WriteImpl(const WriteOptions& write_options,
                          WriteBatch* my_batch, WriteCallback* callback,
                          uint64_t* log_used, uint64_t log_ref,
                          bool disable_memtable) {
+  // The current implementation does not support sync with concurrenet writes
+  assert(!concurrent_writes_ || !write_options.sync);
+  if (concurrent_writes_ && write_options.sync) {
```

Comment:
The name concurrent_writes_ confused me. It feels to me that if it is false, writes cannot be executed concurrently. Can we rename to something more specific, like buffer_wal_writes_in_prepare_ or something like that?